### PR TITLE
refactor(compiler): remove all source-level traces to tsc-wrapped

### PR DESCRIPTION
**Pull request by matsko**

### CODE REVIEW

**Review by tbosch — APPROVED** [review]

File: packages/compiler-cli/test/test_support.ts [hunk 1]
```diff
@@ -25,4 +25,4 @@ export function makeTempDir(): string {
   const dir = path.join(tmpdir, `tmp.${id}`);
   fs.mkdirSync(dir);
   return dir;
-}
+}
```
Comment (IgorMinar) [inline packages/compiler-cli/test/test_support.ts]: keep blank line

**Review by IgorMinar — COMMENTED** [review]
the commit message refers to the packages/tsc-wrapped, but that has been since moved to a separate commit - please update.

otherwise this looks good to me. @tbosch should approve this before merging it since he has the most context for this change.

File: modules/tsconfig-ngc.json [hunk 2]
```diff
@@ -0,0 +1,11 @@
+{
+  "extends": "./tsconfig",
```
Comment (tbosch) [inline modules/tsconfig-ngc.json]: Could you move this into the other wip PR that I should take over (which also contains your start on `test.sh` and `build.sh`?

File: packages/compiler-cli/src/transformers/lower_expressions.ts [hunk 3]
```diff
@@ -35,6 +35,17 @@ function toMap<T, K>(items: T[], select: (item: T) => K): Map<K, T> {
   return new Map(items.map<[K, T]>(i => [select(i), i]));
 }
 
+// copied from typescript core
```
Comment (tbosch) [inline packages/compiler-cli/src/transformers/lower_expressions.ts]: remove this.

File: packages/tsconfig-ngc.json [hunk 4]
```diff
@@ -0,0 +1,11 @@
+{
+  "extends": "./tsconfig",
```
Comment (tbosch) [inline packages/tsconfig-ngc.json]: move into the other wip PR

**Review by tbosch — APPROVED** [review]

File: packages/compiler-cli/src/metadata/bundle_index_host.ts [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+
+  newHost.getSourceFile =
+      (fileName: string, languageVersion: ts.ScriptTarget, onError?: (message: string) => void) => {
+        if (path.normalize(fileName) == normalSyntheticIndexName) {
+          return ts.createSourceFile(fileName, indexContent, languageVersion, true);
+        }
+        return delegate.getSourceFile(fileName, languageVersion, onError);
+      };
+
+  newHost.writeFile =
```
Comment (yogeshgadge) [inline packages/compiler-cli/src/metadata/bundle_index_host.ts]: I believe the writeFile is already defined on WebpackCompilerHost as a getter (not configurable). This is causing https://github.com/angular/angular-cli/issues/8473
`TypeError: Cannot set property writeFile of #<WebpackCompilerHost> which has only a getter`

**Pull request closed without merging**